### DISCUSSION TITLE
Automation account  - Exception calling "Open" with "0" argument(s)

### DISCUSSION THREAD

**Discussion by FestTestQA** [mithrandyr/SimplySql#114] [Q&A]
Hi,

I am trying to execute queries via an Azure automation account runbook.

Executing the query from my local PC works as exected. The same query from AA fails with `Exception calling "Open" with "0" argument(s): "Unable to connect to any of the specified MySQL hosts." (Exception calling "Open" with "0" argument(s): "Unable to connect to any of the specified MySQL hosts." (Unable to connect to any of the specified MySQL hosts. (Timeout expired. The timeout period elapsed prior to completion of the operation or the server is not responding.)))`

Here is the query.

`$password.ToCharArray() | ForEach-Object {$secstr.AppendChar($_)}
$cred = new-object -typename System.Management.Automation.PSCredential -argumentlist $username, $secstr
$srvr = '*******.mysql.database.azure.com'

Open-MySqlConnection -Server $srvr -Database testdb -Credential $cred -CommandTimeout 3000 -SSLMode Required`

**Comment by mithrandyr** [comment]
@FestTestQA -- just out of curiosity, did you verify that the automation account can even reach the MySQL boxes?  That particular error seems to indicate that the MySql server is not reachable by the automation account (i.e. not reaching from whatever server/environment the automation account is running in).
